File: src/main/java/com/talenco/tasksystem/service/ProjectService.java
package com.talenco.tasksystem.service;


import com.talenco.tasksystem.entity.Project;

import java.util.List;

public interface ProjectService {

    List<Project> getAll();

    Project getOne(Long id);

    Long insert(Project project);

    void update(Project project);

    /**
     * 在删除项目时,删除项目下的步骤step,该功能不影响主要供能,可后续完成
     * @param ids 项目id数组
     */
    void delete(Long[] ids);
}

File: src/main/webapp/js/service/projectService.js
app.service('projectService',function($http){

    // 项目列表
    this.getAll=function(){
        return $http.get('../ProjectController/getAll');
    }
    // 项目查询
    this.getOne=function(projectId){
        return $http.get('../ProjectController/getOne?projectId='+projectId);
    }
    // 新建项目
    this.insert=function(entity){
        return $http.post('../ProjectController/insert',entity);
    }
    // 删除项目
    this.dele=function(selectIds){
        if(confirm()){
        return $http.get('../ProjectController/delete?ids='+selectIds);
        }
    }
    // 更新项目信息
    this.update=function(entity){
        return $http.post('../ProjectController/update',entity);
    }
})